### PR TITLE
Features/453146 console cmd support

### PR DESCRIPTION
**Pull request by OlexandrLynda**

### CODE REVIEW

File: src/app/ui/console/console.cpp [hunk 1]
```diff
@@ -31,7 +31,7 @@ Console::Console(QWidget *parent)
     : QPlainTextEdit(parent)
     , localEchoEnabled(false)
 {
-    document()->setMaximumBlockCount(100);
+    document()->setMaximumBlockCount(1000);
```
Comment (Houd1ny) [inline src/app/ui/console/console.cpp]: add setter and getter for this value
it should be member

File: src/app/ui/consoleview.cpp [hunk 2]
```diff
@@ -26,13 +26,17 @@
 #include <QProcess>
 #include <QDebug>
 #include <QDir>
+#include <QSysInfo>
 
 ConsoleView::ConsoleView(QWidget *parent) :
     QWidget(parent),
     ui(new Ui::ConsoleView)
 {
+    mParent = parent;
```
Comment (Houd1ny) [inline src/app/ui/consoleview.cpp]: where do u need this property?
You have to get it from QObject

File: src/app/ui/consoleview.cpp [hunk 3]
```diff
@@ -26,13 +26,17 @@
...
     ui->setupUi(this);
     ui->plainTextEdit->setLocalEchoEnabled(true);
+    ui->plainTextEdit->putData(QByteArray().append("~>"));
+    dirPrinted = true;
```
Comment (Houd1ny) [inline src/app/ui/consoleview.cpp:39]: mDirPrinted

File: src/app/ui/consoleview.cpp [hunk 4]
```diff
@@ -58,13 +62,63 @@ void ConsoleView::execute(QString pCommand)
     mProcess->write(pCommand.toUtf8());
     if(mProcess->atEnd() && !dirPrinted)
     {
-        qDebug() << "dir print from exec";
+        qDebug() << "dir print from exec : " << pCommand;
         slotPrintWorkingDir();
     }
+    if(OsInfo()=="windows")
+    {
+    QProcess* cmd = new QProcess(this); // Link : http://stackoverflow.com/questions/5890955/qprocess-problems-output-of-process
```
Comment (Houd1ny) [inline src/app/ui/consoleview.cpp]: after { tabulation

File: src/app/ui/consoleview.cpp [hunk 5]
```diff
@@ -58,13 +62,63 @@ void ConsoleView::execute(QString pCommand)
...
+    QByteArray resultConsole;
+    QStringList args;
+    //QString cdPathAndCommand = "cd"+consolePath()+"&call "+pCommand;
+    args << "/c"  << pCommand; // << "\r" //Two commands in CMD Link : http://stackoverflow.com/questions/9888806/run-two-commands-in-one-windows-cmd-line-one-command-is-set-command
+                    //need set mPath
+    cmd->start("C:\\Windows\\System32\\cmd", args);
+    //cmd->startDetached("C:\\Program Files\\Git\\bin"); //???need arguments
```
Comment (Houd1ny) [inline src/app/ui/consoleview.cpp]: if u don`t need this comment u should remove it

File: src/app/ui/consoleview.cpp [hunk 6]
```diff
@@ -58,13 +62,63 @@ void ConsoleView::execute(QString pCommand)
...
+        return;
+    }
+
+    cmd->waitForFinished();
+    cmd->waitForReadyRead();
+    cmd->waitForBytesWritten();
+    resultConsole = cmd->readAll();
+    qDebug() << resultConsole;
+
+    if(!resultConsole.isEmpty())
```
Comment (Houd1ny) [inline src/app/ui/consoleview.cpp]: {} codding standart

File: src/app/ui/consoleview.cpp [hunk 7]
```diff
@@ -58,13 +62,63 @@ void ConsoleView::execute(QString pCommand)
...
+    ui->plainTextEdit->putData(" RESULT : " + resultConsole + "~>");
+    //cmd.write("dir\n");
+    //cmd.waitForBytesWritten();
+    //cmd.waitForReadyRead();
```
Comment (Houd1ny) [inline src/app/ui/consoleview.cpp]: remove commented code

File: src/app/ui/consoleview.cpp [hunk 8]
```diff
@@ -58,13 +62,63 @@ void ConsoleView::execute(QString pCommand)
...
+    //qDebug() << cmd.readAll();
+    cmd->waitForFinished();
+    qDebug() << cmd->exitCode();
+
+    }else{
```
Comment (Houd1ny) [inline src/app/ui/consoleview.cpp]: codding standard
else
{
}

File: src/app/ui/consoleview.cpp [hunk 9]
```diff
@@ -58,13 +62,63 @@ void ConsoleView::execute(QString pCommand)
...
+        qDebug() << "Now console view is "
+                    "not support on this system.";
+    }
+
+    //Command inside СonsoleView
+    if(pCommand == "close\n")
+    {
```
Comment (Houd1ny) [inline src/app/ui/consoleview.cpp]: clone shouldn't close application
remove this

File: src/app/ui/consoleview.cpp [hunk 10]
```diff
@@ -58,13 +62,63 @@ void ConsoleView::execute(QString pCommand)
...
+        qDebug() << "close - operation app";
+        qApp->quit();
+    }
+}
+
+//function System Info.
+//To know what in what OS is running SL.
```
Comment (Houd1ny) [inline src/app/ui/consoleview.cpp]: doxygen style comments
/*! enter

File: src/app/ui/consoleview.h [hunk 11]
```diff
@@ -48,6 +48,7 @@ public slots:
 
 private:
     void startProcess();
+    QWidget *mParent;
```
Comment (Houd1ny) [inline src/app/ui/consoleview.h:51]: remove

File: src/app/ui/console/console.cpp [hunk 12]
```diff
@@ -91,7 +89,9 @@ void Console::keyPressEvent(QKeyEvent *e)
 
 void Console::mouseDoubleClickEvent(QMouseEvent *e)
 {
-    Q_UNUSED(e)
+    //Q_UNUSED(e)
```
Comment (Houd1ny) [inline src/app/ui/console/console.cpp]: commented code
remove or not remove
I think it should be there

File: src/app/ui/consoleview.cpp [hunk 13]
```diff
@@ -33,10 +34,12 @@ ConsoleView::ConsoleView(QWidget *parent) :
...
     ui->setupUi(this);
     ui->plainTextEdit->setLocalEchoEnabled(true);
+    ui->plainTextEdit->putData(QByteArray().append("~>"));
+    dirPrinted = true;
     mProcess = new QProcess(this);
     QProcessEnvironment env = QProcessEnvironment::systemEnvironment();
-//    env.insert("PATH", env.value("Path") + ";C:\\Program Files (x86)\\Git\\bin\\;");
-//    env.insert("PATH", env.value("Path") + ";C:\\Program Files (x86)\\Subversion\\bin\\;");
+    //    env.insert("PATH", env.value("Path") + ";C:\\Program Files (x86)\\Git\\bin\\;");
+    //    env.insert("PATH", env.value("Path") + ";C:\\Program Files (x86)\\Subversion\\bin\\;");
```
Comment (Houd1ny) [inline src/app/ui/consoleview.cpp]: there should be commented code
if u need this  code
you shoud add comment why it is needed

**Comment by OlexandrLynda** [comment]
Did all check.

File: src/app/ui/consoleview.cpp [hunk 14]
```diff
@@ -135,10 +208,11 @@ void ConsoleView::startProcess()
 void ConsoleView::slotPrintWorkingDir(QString dir)
 {
     QString lWorkDir = dir;
-//    if(lWorkDir.isEmpty())
-//    {
-//        lWorkDir = QDir::currentPath();
-//    }
-        ui->plainTextEdit->putData(QByteArray().append(lWorkDir+"~>"));
-        dirPrinted = true;
+    //    if(lWorkDir.isEmpty())                   old&not my code.
```
Comment (Houd1ny) [inline src/app/ui/consoleview.cpp:211]: There shouldn`t be commndet code

File: src/app/ui/consoleview.h [hunk 15]
```diff
@@ -39,8 +39,8 @@ class ConsoleView : public QWidget
     explicit ConsoleView(QWidget *parent = 0);
     ~ConsoleView();
     void execute(QString pCommand);
-    QString consolePath();
-
+    QString consolePath(); 
+    QString OsInfo();
```
Comment (Houd1ny) [inline src/app/ui/consoleview.h:43]: const method

File: src/app/ui/consoleview.h [hunk 16]
```diff
@@ -39,8 +39,8 @@ class ConsoleView : public QWidget
     explicit ConsoleView(QWidget *parent = 0);
     ~ConsoleView();
     void execute(QString pCommand);
-    QString consolePath();
-
+    QString consolePath();
```
Comment (Houd1ny) [inline src/app/ui/consoleview.h:42]: const method

File: src/app/ui/consoleview.h [hunk 17]
```diff
@@ -39,8 +39,8 @@ class ConsoleView : public QWidget
     explicit ConsoleView(QWidget *parent = 0);
     ~ConsoleView();
     void execute(QString pCommand);
```
Comment (Houd1ny) [inline src/app/ui/consoleview.h:41]: reference and in all method should be const reference
not copy

File: src/app/ui/consoleview.h [hunk 18]
```diff
@@ -39,8 +39,8 @@ class ConsoleView : public QWidget
...
-    QString consolePath();
-
+    QString consolePath(); 
+    QString OsInfo();
 public slots:
     void slotSetConsolePath(QString pPath);
```
Comment (Houd1ny) [inline src/app/ui/consoleview.h:45]: const reference

File: src/app/ui/consoleview.h (hunk 11 above)
Comment (OlexandrLynda) [inline src/app/ui/consoleview.h:51]: fixed

File: src/app/ui/consoleview.cpp (hunk 14 above)
Comment (OlexandrLynda) [inline src/app/ui/consoleview.cpp:211]: fixed

File: src/app/ui/consoleview.h (hunk 15 above)
Comment (OlexandrLynda) [inline src/app/ui/consoleview.h:43]: fixed

File: src/app/ui/consoleview.h (hunk 17 above)
Comment (OlexandrLynda) [inline src/app/ui/consoleview.h:41]: fixed. func slotExec works only with copy.

File: src/app/ui/consoleview.h (hunk 18 above)
Comment (OlexandrLynda) [inline src/app/ui/consoleview.h:45]: fixed.

**Comment by OlexandrLynda** [comment]
All my changes now are pushed to branch "features/456846_console_sub_project"

**Pull request closed without merging**